infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -897,35 +912,23 @@ private LogEntry getLogEntry(long address, LogData entry) {
     }
 
     /**
-     * Attempts to write a buffer to a file channel, if write fails with an
-     * IOException then the channel pointer is moved back to its original position
-     * before the write
+     * Attempts to write a buffer to a file channel.
      *
      * @param channel the channel to write to
      * @param buf     the buffer to write
      * @throws IOException IO exception
      */
-    private static void safeWrite(FileChannel channel, ByteBuffer buf) throws IOException {
-        long prev = channel.position();
-        try {
-            do {
-                channel.write(buf);
-            } while (buf.hasRemaining());
+    private void writeByteBuffer(FileChannel channel, ByteBuffer buf) throws IOException {
+        int bufSize = buf.remaining();
 
-        } catch (IOException e) {
```

Comment:
We have an outstanding issue where on io exceptions we need to reinitialize the logunit.